Sources/MCP/Server/Resources.swift

```diff
@@ -43,27 +84,90 @@ public struct Resource: Hashable, Codable, Sendable {
+        ) -> Self {
+            .init(uri: uri, mimeType: mimeType, text: content, general: general)
         }
 
-        public static func binary(_ data: Data, uri: String, mimeType: String? = nil) -> Self {
-            .init(uri: uri, mimeType: mimeType, blob: data.base64EncodedString())
+        public static func binary(
+            _ data: Data,
+            uri: String,
+            mimeType: String? = nil,
+            general: GeneralFields = .init()
+        ) -> Self {
+            .init(
+                uri: uri,
+                mimeType: mimeType,
+                blob: data.base64EncodedString(),
+                general: general
+            )
         }
 
-        private init(uri: String, mimeType: String? = nil, text: String? = nil) {
+        private init(
+            uri: String,
+            mimeType: String? = nil,
+            text: String? = nil,
+            general: GeneralFields = .init()
+        ) {
             self.uri = uri
             self.mimeType = mimeType
             self.text = text
             self.blob = nil
+            self.general = general
         }
 
-        private init(uri: String, mimeType: String? = nil, blob: String) {
+        private init(
+            uri: String,
+            mimeType: String? = nil,
+            blob: String,
+            general: GeneralFields = .init()
+        ) {
             self.uri = uri
             self.mimeType = mimeType
             self.text = nil
             self.blob = blob
+            self.general = general
+        }
+
+        private enum CodingKeys: String, CodingKey {
+            case uri
+            case mimeType
+            case text
+            case blob
+        }
+
+        public init(from decoder: Decoder) throws {
+            let container = try decoder.container(keyedBy: CodingKeys.self)
+            uri = try container.decode(String.self, forKey: .uri)
+            mimeType = try container.decodeIfPresent(String.self, forKey: .mimeType)
+            text = try container.decodeIfPresent(String.self, forKey: .text)
+            blob = try container.decodeIfPresent(String.self, forKey: .blob)
+            let dynamic = try decoder.container(keyedBy: DynamicCodingKey.self)
+            general = try GeneralFields.decode(
+                from: dynamic,
+                reservedKeyNames: Self.reservedGeneralFieldNames)
+        }
+
+        public func encode(to encoder: Encoder) throws {
+            var container = encoder.container(keyedBy: CodingKeys.self)
+            try container.encode(uri, forKey: .uri)
+            try container.encodeIfPresent(mimeType, forKey: .mimeType)
+            try container.encodeIfPresent(text, forKey: .text)
+            try container.encodeIfPresent(blob, forKey: .blob)
+            try general.encode(
+                into: encoder,
+                reservedKeyNames: Self.reservedGeneralFieldNames)
+        }
+
+        private static var reservedGeneralFieldNames: Set<String> {
```

Comment:
could `reservedGeneralFieldNames` be a constant?

also rather than hard-coding the same strings that back CodingKeys here, could you instead conform CodingKeys to CaseIterable and write `CodingKeys.allCases.map{ $0.rawValue }` in the variable body?

(these same comments apply to all instances of this property!)